cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -83,7 +83,7 @@ public static void deleteGCSPath(Storage storageClient, String bucket, String pa
       boolean addedToDelete = false;
       for (Blob blob : blobs.iterateAll()) {
         LOG.trace("Added path to be deleted {}", blob.getName());
-        batch.delete(blob.getBlobId());
+        batch.delete(blob.getBlobId(), Storage.BlobSourceOption.generationMatch());
```

Comment:
Why is it needed and how does it work for delete?

Reply:
The delete operation will not be retried by default since it is conditionally idempotent.
It needs the pre-condition `if-GenerationMatch` to be retried.